### PR TITLE
Fix for https://github.com/unclebob/fitnesse/issues/1190

### PR DESCRIPTION
**Pull request by tcnh**
wrapped hash/list conversion when building evaluator context in try/catch block.

### CODE REVIEW

**Comment by fhoeben** [comment]
Can you add a unit test showing the issue reported being handled?

It would, of course, be nicer if no exception would be thrown handing values with a `%` in them...

**Comment by tcnh** [comment]
I agree, but the problem is in org.htmlparser.htmlparser. The parser throws the IndexOutOfBoundsException where I believe it should have just thrown a ParseException.

I don't think we can fix that right now, but I did move the exception handling to where the ParseExceptions are handled and handle the unwanted exception in the same way. I think that makes more sense.

Also added a unit test that feeds the map converter a percentage instead of decent html.

File: src/fitnesse/slim/SlimExpressionEvaluator.java [hunk 1]
```diff
@@ -37,8 +37,8 @@ public void setContext(Map<String, MethodExecutionResult> variables) {
     for (Map.Entry<String, MethodExecutionResult> entry : variables.entrySet()) {
       String key = entry.getKey();
       Object value = entry.getValue().getObject();
-      value = convertWikiHashes(mapCnv, value);
-      value = convertWikiLists(listCnv, value);
+        value = convertWikiHashes(mapCnv, value);
+        value = convertWikiLists(listCnv, value);
```
Comment (fhoeben) [inline src/fitnesse/slim/SlimExpressionEvaluator.java]:
```suggestion
      value = convertWikiLists(listCnv, value);
```

File: src/fitnesse/slim/SlimExpressionEvaluator.java [hunk 2]
```diff
@@ -37,8 +37,8 @@ public void setContext(Map<String, MethodExecutionResult> variables) {
     for (Map.Entry<String, MethodExecutionResult> entry : variables.entrySet()) {
       String key = entry.getKey();
       Object value = entry.getValue().getObject();
-      value = convertWikiHashes(mapCnv, value);
-      value = convertWikiLists(listCnv, value);
+        value = convertWikiHashes(mapCnv, value);
```
Comment (fhoeben) [inline src/fitnesse/slim/SlimExpressionEvaluator.java]:
```suggestion
      value = convertWikiHashes(mapCnv, value);
```

File: test/fitnesse/slim/converters/MapConverterTest.java [hunk 3]
```diff
@@ -134,6 +134,13 @@ public void shouldRenderTableAsHtml() {
         "\t<tr class=\"hash_row\">", "\t\t<td class=\"hash_key\">c</td>", "\t\t<td class=\"hash_value\">d</td>", "\t</tr>", "</table>"), HtmlTag.endl), editor.toString(aMap()));
   }
 
+  @Test
+  public void nonHtmlStringWithPercentageReturnsEmptyMap() {
+    //See https://github.com/unclebob/fitnesse/issues/1190
+    MapConverter cnv = new MapConverter();
+    assertEquals(new HashMap<>(), cnv.fromString("1%"));
```
Comment (fhoeben) [inline test/fitnesse/slim/converters/MapConverterTest.java]: Why don't you use `mapMap()` as done in other tests?

Comment (tcnh) [reply test/fitnesse/slim/converters/MapConverterTest.java]: Agree that's nicer. Didn't pay enough attention to how the other tests are implemented. Current revision is more consistent

**Pull request merged**